app/src/main/kotlin/org/fossify/clock/receivers/AlarmReceiver.kt

```diff
@@ -23,6 +23,8 @@ import org.fossify.commons.helpers.isOreoPlus
 class AlarmReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context, intent: Intent) {
+        context.rescheduleEnabledAlarms();
```

Comment:
that's the important line. The rest is improvements. There's a lot of weird shit in this code base.